### ISSUE TITLE
Recognize : colon - null utility

### ISSUE THREAD

**Issue by covracer** [koalaman/shellcheck#2831]
#### For bugs
- SC1010 and probably others
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:
```sh
# shellcheck shell=bash

t() {
W     : Terraform: Run tfenv then run Terraform with chdir argument
      subcommand=${1:?Specify subcommand as first argument}
      environment=$3
      directory="deploys/$2/terraform"
      shift 3
      tfenv use "$(cat "$directory/.terraform-version")" &&
      (                                                    
          [[ $(terraform -chdir="$directory" workspace show) == "$environment" ]] ||
              terraform -chdir="$directory" workspace select "$environment"         
      ) &&                                                                 
      terraform -chdir="$directory" "$subcommand" "$@"
}
```

#### Here's what shellcheck currently says:
```
Line 4	SC1010: Use semicolon or linefeed before 'then' (or quote to make it literal).
```
#### Here's what I wanted or expected to see:
No errors or warning from line 4, because of the `:` [colon - null utility](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#colon). (Perhaps there's a better way to do it, but I'm trying out [a suggestion](https://stackoverflow.com/a/76949317/3787516) for getting `type t` to include some help text.)
